components/cam/src/physics/cam/froude.F90

```diff
@@ -0,0 +1,176 @@
+    real(r8) :: gam, gammadry, gammawet
+    real(r8) :: es,qs,rh(pver)
+    integer  :: k, kmax, kmin  ! layer index
+    integer  :: i, n,m  ! constituent index
+    real(r8) :: rovg  ! rair/gravit
+    real(r8) :: d
+    integer :: ncol, lchnk
+
+    character(len=*), parameter      :: subname = "froude"
+
+   ncol  = state%ncol
+
+
+   do i=1,ncol
+          do k = 1, pverp
+            pint(k) = hyai(k)*ps0 + hybi(k)*state%ps(i)
+            lnpint(k) = log(pint(k))
+          end do
+          do k = 1, pver
+            pdel(k) = pint(k+1) - pint(k)
+            pmid(k) = 0.5_r8*(pint(k) + pint(k+1))
+            t(k)=state%t(i,k)
+            q(k)=state%q(i,k,1)
+          end do
+          kmax=pver-1
+          do k=kmax-1,1,-1
+             if(pint(kmax)-pint(k).gt.2.e4)then
+                kmin=k
+                exit
+             end if 
+          end do
+
+          do k = 1, pver
+            call qsat(t(k),pmid(k),es,qs)
+            rh(k)=q(k)/qs
+          end do
+!
+!---- determine the brunt-vaisala frequency
+!
+          do k = 2, pver
+            thx(k)=0.5*(state%t(i,k)*(state%ps(i)/pmid(k))**cappa+state%t(i,k-1)*(state%ps(i)/pmid(k-1))**cappa)
+            tint(k)=0.5*(state%t(i,k)+state%t(i,k-1))
+            qint(k)=0.5*(state%q(i,k,1)+state%q(i,k-1,1))
+          end do
+!
+          thx(pver+1) = state%t(i,pver)
+          tint(pver+1) = state%t(i,pver)
+          qint(pver+1) = state%q(i,pver,1)
+!
+          zs = state%phis(i) / gravit
+          zint(pver+1) = zs
+!
+          rovg=rair/gravit
+          do k = pver, 1, -1
+            zint(k) = zint(k+1)+                             &
+                 rovg*state%t(i,k)*(1. + zvir * state%q(i,k,1))*(lnpint(k+1)-lnpint(k))
+          end do
+!
+          do k = 1, pver
+            zmid(k) = 0.5_r8*(zint(k)+zint(k+1))
+          end do
+        !  zmid(pver+1) = zint(pver+1) 
+          zmid(pver) = zint(pver+1) !TKT
+!
+!---- maximum height rise of an air parcel
+!
+          bvfmin = 1.e5_r8
+          do k = kmin,kmax
+            if(rh(k).lt.0.9)then
+              bvf(k) = gravit*(thx(k)-thx(k+1))/         &
+                 (0.5_r8*(thx(k)+thx(k+1))*(zmid(k)-zmid(k+1)))
+            else
+!             moist bvf (Hughes et al., JAS, 2009)
+              call qsat(t(k),pmid(k),es,qs,gam)
+              gammadry=gravit/cpair
+              gammawet=gammadry*(1+q(k))*(1+latvap*qs/(rair*t(k)))/  &
+                 (1+cpwv*qs/cpair+latvap*latvap*epsilo*qs/(cpair*rair*t(k)*t(k))*(1+qs/epsilo))
+              bvf(k) = gravit/t(k)*(gammawet+(tint(k+1)-tint(k))/(zint(k+1)-zint(k)))* &
+                 (1+latvap*qs/(rair*t(k)))-gravit/(1+q(k))*(qint(k+1)-qint(k))/(zint(k+1)-zint(k))
+            endif
+            bvfmin=min(bvf(k),bvfmin)
+
+          end do
+          if(bvfmin <= 0.0_r8) then
+            uovern(i) = 1.e8_r8
+          else
+            uovern(i) = 0.0_r8
+            cmass = 0.0_r8
+            wspdbar=0
+            bvfbar=0
```

Comment:
Both wspbar and bvfbar should be initialized to 0.0_r8

Reply:
Thanks! Requested changes have been applied.